pkg/storage/ingest/reader.go

```diff
@@ -602,50 +602,36 @@ func (r *PartitionReader) WaitReadConsistencyUntilOffset(ctx context.Context, of
 	})
 }
 
-func (r *PartitionReader) waitReadConsistency(ctx context.Context, getOffset func(context.Context) (int64, error)) (returnErr error) {
-	startTime := time.Now()
-	r.metrics.strongConsistencyRequests.Inc()
-
-	spanLog := spanlogger.FromContext(ctx, r.logger)
-	spanLog.DebugLog("msg", "waiting for read consistency")
-
-	// Honor the configured wait timeout.
-	if r.kafkaCfg.WaitStrongReadConsistencyTimeout > 0 {
-		var cancel context.CancelFunc
-		ctx, cancel = context.WithTimeoutCause(ctx, r.kafkaCfg.WaitStrongReadConsistencyTimeout, errWaitStrongReadConsistencyTimeoutExceeded)
-		defer cancel()
-	}
-
-	defer func() {
-		// Do not track failure or latency if the request was canceled (because the tracking would be incorrect).
-		if errors.Is(returnErr, context.Canceled) {
-			return
+func (r *PartitionReader) waitReadConsistency(ctx context.Context, getOffset func(context.Context) (int64, error)) error {
+	_, err := r.metrics.strongConsistencyInstrumentation.Observe(func() (any, error) {
```

Comment:
nit on the signature: `struct{}` instead of `any` would make it more obvious that the return value is ignored

Reply:
Ah good point!

Reply:
Done in https://github.com/grafana/mimir/pull/8808/commits/94d767c523f97d70cffbef4f5f3a3e574b6674e9